packages/terra-functional-testing/tests/jest/commands/utils/ScreenshotRequestor.test.js

```diff
@@ -142,6 +149,160 @@ describe('ScreenshotRequestor', () => {
+      expect(mockInfo).toHaveBeenCalledWith(`Screenshots downloaded from ${screenshotRequestor.url}`);
+    });
+
+    it('calls out an errors', async () => {
+      fetch.mockResolvedValueOnce({
+        ok: true,
+        status: 200,
+      });
+
+      const mockCreateWriteStream = jest.fn();
+      const mockRemoveSync = jest.fn();
+      const mockThrowError = new TypeError("UNKNOWN ERROR");
+      fs.removeSync = mockRemoveSync;
+      fs.createWriteStream = mockCreateWriteStream.mockImplementation( () => {
+        throw mockThrowError;
+      });
+      
+      const oldCheckStatus = ScreenshotRequestor.checkStatus;
+      ScreenshotRequestor.checkStatus = jest.fn();
+      const screenshotRequestor = new ScreenshotRequestor({
+        latestScreenshotsPath: path.join(process.cwd(), 'latest'),
+        referenceScreenshotsPath: path.join(process.cwd(), 'reference'),
+        serviceUrl: 'https://nexus.com/content-compressed/blah/',
+        serviceAuthHeader: 'Basic adfadf',
+        url: 'https://nexus.com/blah/',
+        zipFilePath: path.join(process.cwd(), 'zip-path'),
+      });
+
+      await expect(screenshotRequestor.downloadScreenshots()).rejects.toBe();
+      expect(fetch).toHaveBeenCalledWith(
+        `${screenshotRequestor.serviceUrl}/reference.zip`,
+        {
+          method: 'GET',
+          headers: {
+            Authorization: screenshotRequestor.serviceAuthHeader,
+          },
+        },
+      );
+      expect(ScreenshotRequestor.checkStatus).toHaveBeenCalled();
+      ScreenshotRequestor.checkStatus = oldCheckStatus;
+      expect(mockCreateWriteStream).toHaveBeenCalledWith('terra-wdio-screenshots.zip');
+      expect(mockRemoveSync).toHaveBeenCalledWith('terra-wdio-screenshots.zip');
+      expect(mockError).toHaveBeenCalledWith(`Error occurred while extracting screenshots. ${mockThrowError}`);
+    });
+
+    it('will return early if no screenshots exist', async () => {
+      fetch.mockResolvedValueOnce({
+        ok: true,
+        status: 404,
+      });
+      
+      const screenshotRequestor = new ScreenshotRequestor({
+        latestScreenshotsPath: path.join(process.cwd(), 'latest'),
+        referenceScreenshotsPath: path.join(process.cwd(), 'reference'),
+        serviceUrl: 'https://nexus.com/content-compressed/blah/',
+        serviceAuthHeader: 'Basic adfadf',
+        url: 'https://nexus.com/blah/',
+        zipFilePath: path.join(process.cwd(), 'zip-path'),
+      });
+
+      await screenshotRequestor.downloadScreenshots();
+      expect(fetch).toHaveBeenCalledWith(
+        `${screenshotRequestor.serviceUrl}/reference.zip`,
+        {
+          method: 'GET',
+          headers: {
+            Authorization: screenshotRequestor.serviceAuthHeader,
+          },
+        },
+      );
+      expect(mockInfo).toHaveBeenCalledWith(`No screenshots downloaded from ${screenshotRequestor.url}. Either the URL is invalid or no screenshots were previously uploaded.`);
```

Comment:
Ideally we should have a constants file for all the messages but that can be done in a separate PR. Just making a note for the future.

Reply:
I think the PR is good to merge tight now so let's do it in a separate PR.